Cubical/Algebra/DistLattice/Base.agda

```diff
@@ -0,0 +1,198 @@
+    _∧l_         : A → A → A
+    isDistLattice : IsDistLattice 0l 1l _∨l_ _∧l_
+
+  infix 7 _∨l_
+  infix 6 _∧l_
+
+  open IsDistLattice isDistLattice public
+
+DistLattice : ∀ ℓ → Type (ℓ-suc ℓ)
+DistLattice ℓ = TypeWithStr ℓ DistLatticeStr
+
+isSetDistLattice : (L : DistLattice ℓ) → isSet ⟨ L ⟩
+isSetDistLattice L = L .snd .DistLatticeStr.is-set
+
+makeIsDistLattice : {L : Type ℓ} {0l 1l : L} {_∨l_ _∧l_ : L → L → L}
+             (is-setL : isSet L)
+             (∨l-assoc : (x y z : L) → x ∨l (y ∨l z) ≡ (x ∨l y) ∨l z)
+             (∨l-rid : (x : L) → x ∨l 0l ≡ x)
+             (∨l-comm : (x y : L) → x ∨l y ≡ y ∨l x)
+             (∨l-idem : (x : L) → x ∨l x ≡ x)
+             (∧l-assoc : (x y z : L) → x ∧l (y ∧l z) ≡ (x ∧l y) ∧l z)
+             (∧l-rid : (x : L) → x ∧l 1l ≡ x)
+             (∧l-comm : (x y : L) → x ∧l y ≡ y ∧l x)
+             (∧l-idem : (x : L) → x ∧l x ≡ x)
+             (∧l-absorb-∨l : (x y : L) → x ∧l (x ∨l y) ≡ x)
+             (∧l-ldist-∨l : (x y z : L) → x ∧l (y ∨l z) ≡ (x ∧l y) ∨l (x ∧l z))
+           → IsDistLattice 0l 1l _∨l_ _∧l_
+makeIsDistLattice {_∨l_ = _∨l_} {_∧l_ = _∧l_} is-setL ∨l-assoc ∨l-rid ∨l-comm ∨l-idem
+                                                      ∧l-assoc ∧l-rid ∧l-comm ∧l-idem
+                                                      ∧l-absorb-∨l ∧l-ldist-∨l =
+ isdistlattice (makeIsLattice is-setL ∨l-assoc ∨l-rid (λ x → ∨l-comm _ x ∙ ∨l-rid x) ∨l-comm ∨l-idem
+                                      ∧l-assoc ∧l-rid (λ x → ∧l-comm _ x ∙ ∧l-rid x) ∧l-comm ∧l-idem
+                                      ∨l-absorb-∧l ∧l-absorb-∨l)
+ (λ x y z → ∨l-ldist-∧l _ _ _ , ∨l-rdist-∧l _ _ _) (λ x y z → ∧l-ldist-∨l _ _ _ , ∧l-rdist-∨l _ _ _)
+ where
+ ∨l-absorb-∧l : ∀ x y → x ∨l (x ∧l y) ≡ x
+ ∨l-absorb-∧l x y =
+              cong (_∨l (x ∧l y)) (sym (∧l-idem _)) ∙∙ sym (∧l-ldist-∨l _ _ _) ∙∙ ∧l-absorb-∨l _ _
+
+ ∧l-rdist-∨l : ∀ x y z → (y ∨l z) ∧l x ≡ (y ∧l x) ∨l (z ∧l x)
+ ∧l-rdist-∨l _ _ _ = ∧l-comm _ _ ∙∙ ∧l-ldist-∨l _ _ _ ∙∙ cong₂ (_∨l_) (∧l-comm _ _) (∧l-comm _ _)
+
+ ∨l-ldist-∧l : ∀ x y z → x ∨l (y ∧l z) ≡ (x ∨l y) ∧l (x ∨l z)
+ ∨l-ldist-∧l x y z = x ∨l (y ∧l z)
+                   ≡⟨ cong (_∨l (y ∧l z)) (sym (∨l-absorb-∧l _ _)) ⟩
+                     (x ∨l (x ∧l z)) ∨l (y ∧l z)
+                   ≡⟨ sym (∨l-assoc _ _ _) ⟩
+                     x ∨l ((x ∧l z) ∨l (y ∧l z))
+                   ≡⟨ cong (_∨l ((x ∧l z) ∨l (y ∧l z))) (sym (∧l-comm _ _ ∙ ∧l-absorb-∨l _ _)) ⟩
+                     ((x ∨l y) ∧l x) ∨l ((x ∧l z) ∨l (y ∧l z))
+                   ≡⟨ cong (((x ∨l y) ∧l x) ∨l_) (sym (∧l-rdist-∨l _ _ _)) ⟩
+                     ((x ∨l y) ∧l x) ∨l ((x ∨l y) ∧l z)
+                   ≡⟨ sym (∧l-ldist-∨l _ _ _) ⟩
+                     (x ∨l y) ∧l (x ∨l z) ∎
+
+ ∨l-rdist-∧l : ∀ x y z → (y ∧l z) ∨l x ≡ (y ∨l x) ∧l (z ∨l x)
+ ∨l-rdist-∧l x y z = ∨l-comm _ x ∙∙ ∨l-ldist-∧l _ _ _ ∙∙ cong₂ (_∧l_) (∨l-comm _ _) (∨l-comm _ _)
+
+makeDistLattice : {L : Type ℓ} (0l 1l : L) (_∨l_ _∧l_ : L → L → L)
```

Comment:
That's a good idea, I think we should do that for all the structures in the algebra library. Should that be done now or in a separate PR?

Reply:
Also I think there should be two versions of  `makeIsDistLattice`.
Now only  `∧l-ldist-∨l` has to be provided and then the other distributivity law and `∨l-absorb-∧l` are derived. But it could be helpful to have a version with meet and join swapped.